core/modules/file/views/views_handler_field_file_icon.inc

```diff
@@ -0,0 +1,158 @@
+    $options['add_alt_text'] = array('default' => TRUE);
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function options_form(&$form, &$form_state) {
+    parent::options_form($form, $form_state);
+    $form['icon_size'] = array(
+      '#type' => 'number',
+      '#title' => t('Icon size'),
+      '#min' => 8,
+      '#max' => 300,
+      '#default_value' => $this->options['icon_size'],
+      '#field_suffix' => 'px',
+    );
+    $form['display_type'] = array(
+      '#type' => 'radios',
+      '#title' => t('Display type'),
+      '#options' => array(
+        'svg' => t('Inline SVG markup'),
+        'img' => t('Regular image'),
+      ),
+      '#default_value' => $this->options['display_type'],
+    );
+    $form['add_alt_text'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Add alternative text'),
+      '#default_value' => $this->options['add_alt_text'],
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render($values) {
+    $value = $this->get_value($values);
+
+    // If there is an extension.
+    if ($extension = pathinfo($value, PATHINFO_EXTENSION)) {
+      $icon_name = $this->mapExtensionToName($extension);
+      if ($this->options['display_type'] == 'img') {
+        $icon = $this->renderImg($icon_name, $extension);
+      }
+      else {
+        $icon = $this->renderSvg($icon_name, $extension);
+      }
+      return $icon;
+    }
+  }
+
+  /**
+   * Render the icon as svg tag.
+   */
+  protected function renderSvg($icon_name, $extension) {
+    $options = array(
+      'attributes' => array(
+        'width' => $this->options['icon_size'],
+        'height' => $this->options['icon_size'],
+      ),
+    );
+    if ($this->options['add_alt_text']) {
+      $options['alt'] = t('File icon for @extension extension', array(
+        '@extension' => $extension,
+      ));
+    }
+    return icon($icon_name, $options);
+  }
+
+  /**
+   * Render the icon as img tag.
+   */
+  protected function renderImg($icon_name, $extension) {
+    $attributes = array(
+      'width' => $this->options['icon_size'],
+      'height' => $this->options['icon_size'],
+      'alt' => '',
+      'src' => base_path() . icon_get_path($icon_name),
+    );
+    if ($this->options['add_alt_text']) {
+      $attributes['alt'] = t('File icon for @extension extension', array(
+        '@extension' => $extension,
+      ));
+    }
+    return '<img ' . backdrop_attributes($attributes) . '>';
+  }
+
+  /**
+   * Map file extensions to Phosphor icons.
+   *
+   * @param string $extension
+   *   The original file extension.
+   *
+   * @return string
+   *   The name of an icon file, as expected by icon().
+   */
+  protected function mapExtensionToName($extension) {
+    // If there's no 1:1 icon name match for the extension, map to one. It's
+    // possible that modules or themes provide additional file icons, so this
+    // only acts as a fallback.
+    $mapping = array(
```

Comment:
A suggestion here, we could provide a generic mapping for image/video/archive/text by using `file_get_mimetype()`, which would return `image/gif`, `video/x-msvideo`, or similar. Then pull off the leading part to get `image`, `video`, etc. Then use the manual map for extensions with a specific icon.

Reply:
In reworked this to use the mimetype mapping as described.